### PR TITLE
feat: validate duplicate provider names in NewRequestBridge

### PR DESCRIPTION
**Pull request by ssncferreira**
## Description

Adds validation to `NewRequestBridge` to prevent registering multiple providers with the same name. Duplicate names would cause route prefix conflicts and ambiguous metrics, logs, and circuit breaker behavior.

## Changes

* Add `validateProviders()` that rejects duplicate provider names at startup
* Add table-driven tests covering valid configurations and duplicate name detection

Related to: https://github.com/coder/aibridge/issues/152

_Disclaimer: initially produced by Claude Opus 4.6, heavily modified and reviewed by @ssncferreira ._

### CODE REVIEW

**Comment by ssncferreira** [comment]
* **#240** <a href="https://app.graphite.com/github/pr/coder/aibridge/240?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/coder/aibridge/240?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#238** <a href="https://app.graphite.com/github/pr/coder/aibridge/238?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#237** <a href="https://app.graphite.com/github/pr/coder/aibridge/237?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: bridge.go [hunk 1]
```diff
@@ -57,6 +57,18 @@ type RequestBridge struct {
...
+// validateProviders checks that no two providers share the same name.
+func validateProviders(providers []provider.Provider) error {
+	names := make(map[string]bool, len(providers))
+	for _, prov := range providers {
+		if names[prov.Name()] {
+			return fmt.Errorf("duplicate provider name: %q", prov.Name())
+		}
+		names[prov.Name()] = true
+	}
+	return nil
```
Comment (ssncferreira) [inline bridge.go:77]: Initially, I added a check to confirm that there are no 2 providers with the same base URL. However, this was failing on the tests, because `newBridgeTestServer` receives a single upstreamURL, which is then used for both anthropic and openai: https://github.com/coder/aibridge/blob/main/internal/integrationtest/setupbridge.go#L163-L164

We might need to update the tests, but that would require a bigger refactor.

Comment (pawbana) [reply bridge.go:77]: Maybe I'm missing something but I don't think we should care about base url being the same?

The only issue is prefix provider being mounted on being the same which is checked.

I could imagine this being useful, eg. having the same provider with the same base URL (default value) with different keys behind them and different permissions to them. Something like per org provider of the same type/upstream with different limits.

Comment (ssncferreira) [reply bridge.go:77]: That is a good point, I didn't think of it. Removed the TODO: https://github.com/coder/aibridge/pull/240/commits/225fdde21903935b8e234e7e61e82ccf2ff8cf64

**Review by dannykopping — APPROVED** [review]
The timing for this is funny because I have to look into https://linear.app/codercom/issue/AIGOV-157/spike-to-understand-if-there-is-a-simple-way-to-handle-multi-api-key which may _require_ multiple providers of the same name, but I'll have to find a way around this.

**Review by pawbana — APPROVED** [review]

**Comment by ssncferreira** [comment]
### Merge activity

* **Mar 31, 2:32 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/coder/aibridge/240).
* **Mar 31, 2:36 PM UTC**: [Graphite](https://app.graphite.com/github/pr/coder/aibridge/240) rebased this pull request as part of a merge.
* **Mar 31, 2:37 PM UTC**: @ssncferreira merged this pull request with [Graphite](https://app.graphite.com/github/pr/coder/aibridge/240).
